bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java

```diff
@@ -115,6 +171,11 @@ public void added(ItemChannelLink itemChannelLink) {
 
         @Override
         public void removed(ItemChannelLink itemChannelLink) {
+            if (itemRegistry.get(itemChannelLink.getItemName()) == null) {
+                // Don't inform about the unlink if the item does not exist.
+                // The handler is informed if the item has been removed.
+                return;
```

Comment:
> I would be fine to actually only call channelUnlinked() when a link is removed and don't care about items.

makes sense. we should be aware though that bindings might get two "linked" events without a "unlinked" in between, if just the item dis- and reappears for whatever reason. Would be okay as far as I can see, but should be documented.